asf_heat_pump_suitability/pipeline/run_scripts/run_get_inspire_file_bounds.py

```diff
@@ -0,0 +1,89 @@
+"""
+Generate bounding polygons of each INSPIRE land registry file and save to S3.
+
+Land registry files are partitioned by Registration County in Scotland and by local authority in England and Wales. To
+match land registry files with Microsoft building footprint files, we first must identify the bounding geometry of each
+land registry file.
+
+Scotland INSPIRE files are smaller (they are shapefiles) and there are fewer (33). They are loaded individually in this
+script and the bounding polygon generated.
+
+England and Wales INSPIRE files are larger (.gml files) and there are more (300+). They are matched to existing local
+authority boundaries via string matching based on their filenames. Any nulls are filled by loading the land registry
+file and identifying which local authority boundary it should be matched to based on the geometry.
+
+To run:
+python -i asf_heat_pump_suitability/pipeline/run_scripts/run_get_inspire_file_bounds.py -y [YYYY] -q [N] -n all
+
+[Set -n nation flag to "ew" or "s" for generating file bounds either England and Wales or Scotland INSPIRE files only.]
+"""
+
+import pandas as pd
+import argparse
+from argparse import ArgumentParser
+from asf_heat_pump_suitability import config
+from asf_heat_pump_suitability.pipeline.prepare_features import (
+    land_extent,
+)
+
+
+def parse_arguments() -> argparse.Namespace:
+    """
+    Create ArgumentParser and parse.
+
+    Returns:
+        argparse.Namespace: populated `Namespace`
+    """
+    parser = ArgumentParser()
+
+    parser.add_argument(
+        "-y",
+        "--year",
+        help="EPC data year. Format YYYY",
+        type=int,
+        required=True,
+    )
+
+    parser.add_argument(
+        "-q",
+        "--quarter",
+        help="EPC data quarter",
+        type=int,
+        required=True,
+    )
+
+    parser.add_argument(
+        "-n",
+        "--nations",
+        help="Nations to get INSPIRE land registry file bounds for. Of England and Wales (ew); Scotland (s); or all (ews).",
+        type=str,
+        choices=["ew", "s", "all"],
+        required=True,
+    )
+
+    return parser.parse_args()
+
+
+if __name__ == "__main__":
```

Comment:
We do have an existing version of the file bounds for England and Wales, it just has a different name at the moment: `2023_land_parcels_with_file_polygons.geojson`. I need to run this script for England and Wales so that it produces outputs with the expected names. I just hadn't done it yet because the `_fill_null_file_bounds` function takes a long time to run and I was focussing on Scotland.

Currently, `run_scripts/run_calculate_garden_size.py` will fail if running for England and Wales because it will look for a mapping file with a different name. This can be updated in the script. I will run this `run_get_inspire_file_bounds.py` for England and Wales now though to generate the appropriate file and for testing purposes.

Reply:
The `inspire_file_bounds_ew.geojson` is now available in s3.